pkg/security/ebpf/c/raw_syscalls.h

```diff
@@ -0,0 +1,154 @@
+#ifndef _RAW_SYSCALLS_H
+#define _RAW_SYSCALLS_H
+
+#include "defs.h"
+
+struct _tracepoint_raw_syscalls_sys_enter {
+    unsigned short common_type;
+    unsigned char common_flags;
+    unsigned char common_preempt_count;
+    int common_pid;
+    long id;
+    unsigned long args[6];
+};
+
+struct bpf_map_def SEC("maps/concurrent_syscalls") concurrent_syscalls = {
+    .type = BPF_MAP_TYPE_ARRAY,
+    .key_size = sizeof(u32),
+    .value_size = sizeof(long),
+    .max_entries = 1,
+    .pinning = 0,
+    .namespace = "",
+};
+
+struct bpf_map_def SEC("maps/sys_exit_progs") sys_exit_progs = {
+    .type = BPF_MAP_TYPE_PROG_ARRAY,
+    .key_size = sizeof(u32),
+    .value_size = sizeof(u32),
+    .max_entries = 64,
+};
+
+#define CONCURRENT_SYSCALLS_COUNTER 0
+
+struct process_syscall_t {
+    char comm[TASK_COMM_LEN];
+    int pid;
+    int id;
+};
+
+struct bpf_map_def SEC("maps/noisy_processes_fb") noisy_processes_fb = {
+    .type = BPF_MAP_TYPE_LRU_HASH,
+    .key_size = sizeof(struct process_syscall_t),
+    .value_size = sizeof(u64),
+    .max_entries = 2048,
+    .pinning = 0,
+    .namespace = "",
+};
+
+struct bpf_map_def SEC("maps/noisy_processes_bb") noisy_processes_bb = {
+    .type = BPF_MAP_TYPE_LRU_HASH,
+    .key_size = sizeof(struct process_syscall_t),
+    .value_size = sizeof(u64),
+    .max_entries = 2048,
+    .pinning = 0,
+    .namespace = "",
+};
+
+struct bpf_map_def SEC("maps/kill_list") kill_list = {
+    .type = BPF_MAP_TYPE_LRU_HASH,
+    .key_size = sizeof(u32),
+    .value_size = sizeof(u32),
+    .max_entries = 256,
+    .pinning = 0,
+    .namespace = "",
+};
+
+SEC("tracepoint/raw_syscalls/sys_enter")
+int sys_enter(struct _tracepoint_raw_syscalls_sys_enter *args) {
+    u64 signal_processes;
+    u32 pid = bpf_get_current_pid_tgid() >> 32;
+    u32 *sig = bpf_map_lookup_elem(&kill_list, &pid);
```

Comment:
Should we remove the pid from the kill list when we got an entry to avoid possible signal storms if the signal sent won't terminate the process (like SIGUSR1) ?

Reply:
Indeed, done